.github/workflows/build.yml

```diff
@@ -13,38 +13,71 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        name: [fedora-gcc, fedora-clang, debian-clang, fedora-distcheck]
+        name: [fedora-softokn-gcc, fedora-softokn-clang, debian-softokn-clang, fedora-softokn-distcheck,
```

Comment:
You should probably use a multidimensional matrix here instead of manually expanding the names:

Define:
token: [softokn, softhsm]
name: [fedora-gcc, fedora-clang, debian-clang, fedora-distcheck]

This will automatically create all 8 jobs

then in use if [ "${{ matrix.token }}" in the tests to know what to install